x/ecocredit/server/basket/create.go

```diff
@@ -3,10 +3,41 @@ package basket
 import (
 	"context"
 
-	baskettypes "github.com/regen-network/regen-ledger/x/ecocredit/basket"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	basketv1 "github.com/regen-network/regen-ledger/api/regen/ecocredit/basket/v1"
+	"github.com/regen-network/regen-ledger/x/ecocredit/basket"
 )
 
-func (k Keeper) Create(ctx context.Context, create *baskettypes.MsgCreate) (*baskettypes.MsgCreateResponse, error) {
-	//TODO implement me
-	panic("implement me")
+func (k Keeper) Create(ctx context.Context, msg *basket.MsgCreate) (*basket.MsgCreateResponse, error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	fee := k.ecocreditKeeper.GetCreateBasketFee(ctx)
+	if err := msg.Validate(fee); err != nil {
+		return nil, err
+	}
+	// TODO: need to decide about the denom creation
+	denom := msg.Name
+	sender, err := sdk.AccAddressFromBech32(msg.Curator)
+	if err != nil {
+		return nil, err
+	}
+
+	err = k.stateStore.BasketStore().Insert(ctx, &basketv1.Basket{
+		BasketDenom:       denom,
+		DisableAutoRetire: msg.DisableAutoRetire,
+		CreditTypeName:    msg.CreditTypeName,
+		DateCriteria:      msg.DateCriteria.ToApi(),
+		Exponent:          msg.Exponent,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	k.bankKeeper.SendCoinsFromAccountToModule(sdkCtx, sender, feeModuleAccName, fee)
```

Comment:
is this better than burning?

Reply:
okay, let's keep as is

Reply:
I don't think we can do this this way as it will break invariants.

Funding the community account should (I'm pretty sure) go through `distributionKeeper.FundCommunityPool()`, as the distribution keeper also tracks internally the balance of the community pool independently from the actual balance in the module account.